File: Python/05 DFSBFS/게임맵최단거리.py
from collections import deque

dx = [-1, 1, 0 ,0]
dy = [0, 0, -1, 1]    

def solution(maps):
    answer = 0
    queue = deque()
    queue.append([0, 0])
    n = len(maps)
    m = len(maps[0])
    while queue:
        x, y = queue.popleft()
        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]

            if (0<= nx < n) and (0 <= ny < m) and maps[nx][ny] == 1:
                queue.append([nx,ny])
                maps[nx][ny] = maps[x][y] + 1
    
    answer = maps[n-1][m-1]
    if answer == 1: return -1
    else: return maps[n-1][m-1]

File: JavaScript/BFS/게임맵최단거리.js
function solution(maps) {
  var answer = 1;
  const dx = [-1, 1, 0, 0];
  const dy = [0, 0, 1, -1];
  var queue = [[0, 0]];
  const n = maps.length;
  const m = maps[0].length;
  console.log(n, m);
  while (queue.length > 0) {
    const [x, y] = queue.shift();
    for (let i = 0; i < 4; i++) {
      const nx = x + dx[i];
      const ny = y + dy[i];
      if (nx >= 0 && nx < n && ny >= 0 && ny < m) {
        if (maps[nx][ny] === 1) {
          queue.push([nx, ny]);
          maps[nx][ny] = maps[x][y] + 1;
        }
      }
    }
  }

  return maps[n - 1][m - 1] === 1 ? -1 : maps[n - 1][m - 1];
}
